app/controllers/nonprofits_controller.rb

```diff
@@ -55,10 +55,20 @@ def create
   end
 
   def update
-    flash[:notice] = "Update successful!"
-    current_nonprofit.update_attributes params[:nonprofit].except(:verification_status)
-    current_nonprofit.clear_cache
-    json_saved current_nonprofit
+    return render json: ["Not authorized"], status: 403 unless current_role?([:nonprofit_admin, :super_admin])
```

Comment:
This isn't quite right. The nonprofit associate CAN update the nonprofit object but they can't update certain attributes on the object.